drivers/ethernet/eth_lan865x.c

```diff
@@ -410,6 +411,9 @@ static void lan865x_read_chunks(const struct device *dev)
 		return;
 	}
 
+	/* Try to keep headers compact */
+	net_pkt_compact(pkt);
```

Comment:
We should avoid calling this, as it causes lot of memcpy.
Better would be to change the oa_tc6.c to read data to temp buffer and then use `net_pkt_write()` to write the data to the pkt. This way the data is contiguous from the start and there is no need to compact it.